pkg/sql/importer/exportparquet_test.go

```diff
@@ -143,84 +144,130 @@ func validateParquetFile(
 	dbName := "rand"
-
-	params := base.TestClusterArgs{
-		ServerArgs: base.TestServerArgs{
-			UseDatabase:   dbName,
-			ExternalIODir: dir,
-		},
-	}
+	srv, db, _ := serverutils.StartServer(t, base.TestServerArgs{
+		UseDatabase:   dbName,
+		ExternalIODir: dir,
+	})
 	ctx := context.Background()
-	tc := testcluster.StartTestCluster(t, 3, params)
-	defer tc.Stopper().Stop(ctx)
-	sqlDB := sqlutils.MakeSQLRunner(tc.Conns[0])
-
+	defer srv.Stopper().Stop(ctx)
+	sqlDB := sqlutils.MakeSQLRunner(db)
+	rng, _ := randutil.NewTestRand()
 	sqlDB.Exec(t, fmt.Sprintf("CREATE DATABASE %s", dbName))
 
-	tableName := "table_1"
-	// TODO (butler): Randomly generate additional table(s) using tools from PR
-	// #75677. The function below only creates a deterministic table with a bunch
-	// of interesting corner case values.
-	err := randgen.GenerateRandInterestingTable(tc.Conns[0], dbName, tableName)
-	require.NoError(t, err)
-
-	s0 := tc.Server(0)
-	ie := s0.ExecutorConfig().(sql.ExecutorConfig).InternalExecutor
+	var tableName string
+	ie := srv.ExecutorConfig().(sql.ExecutorConfig).InternalExecutor
+	// Try at most 10 times to populate a random table with at least 10 rows.
 	{
-		// Ensure table only contains columns supported by EXPORT Parquet
-		_, cols, err := ie.QueryRowExWithCols(
-			ctx,
-			"",
-			nil,
-			sessiondata.InternalExecutorOverride{
-				User:     security.RootUserName(),
-				Database: dbName,
-			},
-			fmt.Sprintf("SELECT * FROM %s LIMIT 1", tableName))
-		require.NoError(t, err)
-
-		for _, col := range cols {
-			_, err := importer.NewParquetColumn(col.Typ, "", false)
-			if err != nil {
-				t.Logf("Column type %s not supported in parquet, dropping", col.Typ.String())
-				sqlDB.Exec(t, fmt.Sprintf(`ALTER TABLE %s DROP COLUMN %s`, tableName, col.Name))
+		var (
+			i           int
+			success     bool
+			tablePrefix = "table"
+			numTables   = 20
+		)
+
+		stmts := randgen.RandCreateTables(rng, tablePrefix, numTables,
+			randgen.PartialIndexMutator,
+			randgen.ForeignKeyMutator,
+		)
+
+		var sb strings.Builder
+		for _, stmt := range stmts {
+			sb.WriteString(tree.SerializeForDisplay(stmt))
+			sb.WriteString(";\n")
+		}
+		sqlDB.Exec(t, sb.String())
+
+		for i = 1; i < numTables; i++ {
+			tableName = tablePrefix + fmt.Sprint(i)
+			numRows, err := randgen.PopulateTableWithRandData(rng, db, tableName, 20)
+			require.NoError(t, err)
+			if numRows > 5 {
+				// Ensure the table only contains columns supported by EXPORT Parquet. If an
+				// unsupported column cannot be dropped, try populating another table
+				if err := func() error {
+					_, cols, err := ie.QueryRowExWithCols(
+						ctx,
+						"",
+						nil,
+						sessiondata.InternalExecutorOverride{
+							User:     security.RootUserName(),
+							Database: dbName},
+						fmt.Sprintf("SELECT * FROM %s LIMIT 1", tableName))
+					require.NoError(t, err)
+
+					for _, col := range cols {
+						_, err := importer.NewParquetColumn(col.Typ, "", false)
```

Comment:
This doesn't need to block this PR. But I wonder if we could extend RandCreateTables to take some sort of type filter so that it doesn't produce columns of the given types.  Then in the importer package we could have a finite list of types we don't support and this test would assume that all others are supported.  It would both have the benefit of cleaning up this test a bit and also ensuring that as new types are added, we default to assuming they should work.

Reply:
Good idea. I'll file an issue.